src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/rca/pyrometer/Api.java

```diff
@@ -0,0 +1,152 @@
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import javax.annotation.Nonnull;
+
+/**
+ * Although Pyrometer exists on all the nodes - data and master but the APIs are only accessible
+ * from the master nodes. Pyrometer is able answer queries such as:
+ * - All nodes by in Zone <i>Z</i> by dimension <i>D</i> order by <i>Temperature</i> in
+ * descending order.
+ * - The top <i>K</i> warm shards on the hottest nodes by dimension D over a period of h hours
+ * -
+ */
+public class Api {
+    enum Count {
+        /**
+         * This is to ask the API to return all the contenders.
+         */
+        ALL,
+
+        /**
+         * This is to return the top K contenders
+         */
+        TOP_K
+    }
+
+    enum SortOrder {
+        ASCENDING,
+        DESCENDING;
+    }
+
+    /**
+     * Api class should be called only from elected master node.
+     *
+     * @return true if elected master or false otherwise.
+     */
+    private static boolean checkIfElectedMaster() {
+        return true;
+    }
+
+    private static void failIfNotElectedMaster(String apiName) {
+        if (!checkIfElectedMaster()) {
+            StringBuilder builder = new StringBuilder();
+            builder.append("Api (").append(apiName).append(") can only be called from the elected"
+                    + " master node.");
+
+            throw new IllegalStateException(builder.toString());
+        }
+    }
+
+    /**
+     * A cluster is considered imbalanced along a dimension if the temperature of the hottest
+     * node and the coldest node are too far apart. How far is too far, is something to be
+     * evaluated.
+     *
+     * @param dimension The dimension to consider.
+     * @return The difference in temperature of the
+     */
+    public static boolean isClusterImbalanceAlongDimension(TemperatureVector.Dimension dimension) {
+        failIfNotElectedMaster("getClusterImbalanceAlongDimension");
+        // TODO: Calculate this
+        return true;
+    }
+
+
+    /**
+     * To get the list of all the nodes in the cluster order.
+     *
+     * @param dimension Temperature along this dimension will be considered to sort the nodes
+     * @param zone      Only the nodes in this zone will be reported.
+     * @param count     How many nodes to return in the result.
+     * @param order     Will the nodes be sorted in the increasing order or decreasing order.
+     * @return An ordered list of nodes.
+     */
+    public static @Nonnull
+    List<CompactNodeSummary> getNodesForGivenZone(final TemperatureVector.Dimension dimension,
+                                                  final HeatZoneAssigner.Zone zone,
+                                                  final Count count,
+                                                  final SortOrder order) {
+        failIfNotElectedMaster("getNodes");
+        return new ArrayList<>();
```

Comment:
I am assuming this is a WIP since we are only asserting the master criterion and returning an empty arraylist here.

I think it would be good to make description of the PR more verbose to indicate this includes WIP items and what they are.